superset-frontend/src/dataMask/types.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ExtraFormData, DataMaskCurrentState } from '@superset-ui/core';
+
+export enum DataMaskType {
+  NativeFilters = 'nativeFilters',
+  CrossFilters = 'crossFilters',
+  OwnFilters = 'ownFilters',
+}
+
+export type SingleMask = {
+  extraFormData?: ExtraFormData;
+  currentState: DataMaskCurrentState;
+};
+
+export type MultipleMask = SingleMask & { id: string };
+
+export type MultipleDataMask = { [filterId: string]: MultipleMask };
+export type MultipleDataMaskState = {
+  [DataMaskType.NativeFilters]: MultipleDataMask;
+  [DataMaskType.CrossFilters]: MultipleDataMask;
+  [DataMaskType.OwnFilters]: MultipleDataMask;
+};
+
+export type SingleDataMask = { [filterId: string]: SingleMask };
+export type SingleDataMaskState = {
+  [DataMaskType.NativeFilters]: SingleMask;
+  [DataMaskType.CrossFilters]: SingleMask;
+  [DataMaskType.OwnFilters]: SingleMask;
+};
```

Comment:
These feel slightly confusing. I think something like this could help clarify for the casual developer that stumbles into this area of the code:

- Instead of `SingleMask` we would have `DataMask`
- Instead of `MultipleDataMask` we would have `DataMaskWithId`
- Instead of `MultipleDataMaskState` we would have `DataMaskStateWithId` (not thrilled about this name, but it's still slightly more descriptive IMO)
- Instead of `SingleDataMaskState` we would then have `DataMaskState`

Reply:
A little bit changed on other names